### PR TITLE
CLN: ensure_platform_int earlier

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

**Comment by jreback** [comment]
hmm 32-bit is failing here (one of the posix failed too, but that is unrelated i think)

File: pandas/core/arrays/categorical.py [hunk 1]
```diff
@@ -1980,9 +1980,9 @@ def _reverse_indexer(self) -> Dict[Hashable, np.ndarray]:
         """
         categories = self.categories
         r, counts = libalgos.groupsort_indexer(
-            self.codes.astype("int64", copy=False), categories.size
+            libalgos.ensure_platform_int(self.codes), categories.size
```
Comment (simonjayhawkins) [inline pandas/core/arrays/categorical.py:1983]: IIUC ensure_platform_int should be imported from pandas.core.dtypes.common see #40567

Comment (jbrockmendel) [reply pandas/core/arrays/categorical.py:1983]: sure, mind if i do this in the next pass?  have a branch on deck

Comment (jreback) [reply pandas/core/arrays/categorical.py:1983]: ok to do later, but maybe should rebase this PR anyhow as @simonjayhawkins change was merged

Comment (jbrockmendel) [reply pandas/core/arrays/categorical.py:1983]: sure

**Review by jreback — APPROVED** [review]
lgtm ping on green

**Comment by jbrockmendel** [comment]
Ping

**Pull request merged**